tools/bin/hawqregister

```diff
@@ -126,182 +127,319 @@ def option_parser_yml(yml_file):
+        else:
+            self.file_format = 'Parquet'
+            check_hash_type() # Usage1 only support randomly distributed table
+        if not self.filepath:
+            sys.exit(0)
 
-    try:
-        # get the full path of correspoding file for target table
-        query = ("select location, gp_persistent_tablespace_node.tablespace_oid, database_oid, relfilenode from pg_class, gp_persistent_relation_node, "
-                 "gp_persistent_tablespace_node, gp_persistent_filespace_node where relname = '%s' and pg_class.relfilenode = "
-                 "gp_persistent_relation_node.relfilenode_oid and gp_persistent_relation_node.tablespace_oid = gp_persistent_tablespace_node.tablespace_oid "
-                 "and gp_persistent_filespace_node.filespace_oid = gp_persistent_filespace_node.filespace_oid;") % tablename.split('.')[-1]
-        conn = dbconn.connect(dburl, False)
-        rows = dbconn.execSQL(conn, query)
-        conn.commit()
-        conn.close()
-    except DatabaseError, ex:
-        logger.error('Failed to execute query "%s"' % query)
-        sys.exit(1)
-    for row in rows:
-        tabledir = '/'.join([row[0].strip(), str(row[1]), str(row[2]), str(row[3]), ''])
-    return firstsegno, tabledir
-
-
-def check_files_and_table_in_same_hdfs_cluster(filepath, tabledir):
-    '''Check whether all the files refered by 'filepath' and the location corresponding to the table are in the same hdfs cluster'''
-    if not filepath:
-        return
-    # check whether the files to be registered is in hdfs
-    filesystem = filepath.split('://')
-    if filesystem[0] != 'hdfs':
-        logger.error('Only support to register file(s) in hdfs')
-        sys.exit(1)
-    fileroot = filepath.split('/')
-    tableroot = tabledir.split('/')
-    # check the root url of them. eg: for 'hdfs://localhost:8020/temp/tempfile', we check 'hdfs://localohst:8020'
-    if fileroot[0:3] != tableroot[0:3]:
-        logger.error("Files to be registered and the table are not in the same hdfs cluster.\nFile(s) to be registered: '%s'\nTable path in HDFS: '%s'" % (filepath, tabledir))
-        sys.exit(1)
+        self.seg_name = get_seg_name()
+        self.firstsegno, self.tabledir = get_metadata()
 
+        if self.mode == 'repair':
+            if self.tabledir.strip('/') != self.filepath.strip('/'):
+                logger.error('In repair mode, tablename in yml file should be the same with input args')
+                sys.exit(1)
+            check_policy_consistency()
+            check_bucket_number()
+            existed_files, existed_sizes = self._get_files_in_hdfs(self.filepath)
+            existed_info = {}
+            for k, fn in enumerate(existed_files):
+                existed_info[fn] = existed_sizes[k]
+            for k, fn in enumerate(self.files):
+                if fn not in existed_files:
+                    logger.error('Can not register --repair since giving non-existed files.')
+                    sys.exit(1)
```

Comment:
Could we print more detail information for non-existed files? Also change --repair to repair mode.